funnel/utils/markdown/tabs.py

```diff
@@ -0,0 +1,245 @@
+"""MDIT renderer and helpers for tabs."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from functools import reduce
+from typing import ClassVar, Dict, List, Optional, Tuple
+
+__all__ = ['render_tab']
+
+
+def get_tab_tokens(tokens):
+    return [
+        {
+            'index': i,
+            'nesting': token.nesting,
+            'info': token.info,
+            'key': '-'.join([token.markup, str(token.level)]),
+        }
+        for i, token in enumerate(tokens)
+        if token.type in ('container_tab_open', 'container_tab_close')
+    ]
+
+
+def get_pair(tab_tokens, start=0) -> Optional[Tuple[int, int]]:
```

Comment:
`tab_tokens` needs a type hint. `start` is presumed to be `int`, but could do with an explicit hint.